src/obb_rules/serializer/writer.cljc

```diff
@@ -0,0 +1,62 @@
+(ns ^{:added "2.1.0"
+      :author "Pedro Santos"}
+  obb-rules.serializer.writer
+  "Writes a game and actions to a text format"
+  (:require [obb-rules.game :as game]
+            [obb-rules.game-mode :as game-mode]))
+
+(defmulti action->str
+  "Translates a raw action to a concise string representation"
+  (fn [action]
+    (first action)))
+
+(defmethod action->str :deploy
+  [[_ quantity unit [source-x source-y]]]
+  (str "d" source-x source-y "." quantity "." (name unit)))
+
+(defmethod action->str :auto-deploy
+  [[_ auto-template]]
+  (str "auto-" (name auto-template)))
+
+(defmethod action->str :attack
+  [[_ [source-x source-y] [target-x target-y]]]
+  (str "a" source-x source-y target-x target-y))
+
+(defmethod action->str :goto
+  [[_ [source-x source-y] [target-x target-y]]]
+  (str "g" source-x source-y target-x target-y))
+
+(defmethod action->str :rotate
+  [[_ [source-x source-y] dir]]
+  (str "r" source-x source-y (first (name dir))))
+
+(defmethod action->str :move
+  [[_ [source-x source-y] [target-x target-y] quantity]]
+  (str "m" source-x source-y target-x target-y "." quantity))
+
+(defn actions->str
+  "Translates a coll of raw actions to a concise string representation"
+  [actions]
+  (->> actions
+       (map action->str)
+       (clojure.string/join " ")))
+
+(def separator "\n\n")
```

Comment:
Would it make sense to be more specific in this name? I'd argue that the space used to separate action on `actions->str` is also a separator (maybe it is also worth it to extract that one into a constant?). I'd also suggest that you add metadata to signal that these values are constants.

Maybe I am nitpicking again and I understand if you don't think the changes are worth it.